modules/internal/src/main/scala/service.scala

```diff
@@ -273,24 +262,24 @@ private[internal] object utils {
   // format: OFF
   def buildRequests(algName: Type.Name, typeParam: Type.Param, stats: List[Stat]): List[RPCRequest] = stats.collect {
     case q"@rpc($s) @stream[ResponseStreaming.type] def $name[..$tparams]($request): $typeParam[Observable[$response]]" =>
-      Option(RPCRequest(algName, name, utils.serializationType(s), Some(ResponseStreaming), paramTpe(request), response))
+      RPCRequest(algName, name, utils.serializationType(s), Some(ResponseStreaming), paramTpe(request), response)
     case q"@rpc($s) @stream[RequestStreaming.type] def $name[..$tparams]($paranName: Observable[$request]): $typeParam[$response]" =>
-      Option(RPCRequest(algName, name, utils.serializationType(s), Some(RequestStreaming), request, response))
+      RPCRequest(algName, name, utils.serializationType(s), Some(RequestStreaming), request, response)
     case q"@rpc($s) @stream[BidirectionalStreaming.type] def $name[..$tparams]($paranName: Observable[$request]): $typeParam[Observable[$response]]" =>
-      Option(RPCRequest(algName, name, utils.serializationType(s), Some(BidirectionalStreaming), request, response))
+      RPCRequest(algName, name, utils.serializationType(s), Some(BidirectionalStreaming), request, response)
     case q"@rpc($s) def $name[..$tparams]($request): $typeParam[$response]" =>
-      Option(RPCRequest(algName, name, utils.serializationType(s), None, paramTpe(request), response))
-    case e =>
-      None
-  }.flatten
+      RPCRequest(algName, name, utils.serializationType(s), None, paramTpe(request), response)
+  }
```

Comment:
Doing this, @service combined with @tagless might not work, since tagless macro is adding a `mapK` which is not being considered by this @service macro.

Reply:
In what way did I change anything here?

Isn't `(xs.collect(f): List[Option[B]]).flatten` the same as `xs.collect(g): List[B]`?

Reply:
Correct, sorry, I missed it :(